File: java/src/test/java/problems1_100/PlusOneTest.java
package problems1_100;

import java.util.Arrays;
import org.junit.Test;
import sample.SampledTest;

import static com.google.common.truth.Truth.assertWithMessage;

public class PlusOneTest extends SampledTest {
    @Test
    public void test() {
        for (PlusOne solution : PlusOne.values()) {
            for (PlusOneSample sample : getSamples(PlusOneSample[].class)) {
                assertWithMessage(sample.getMessage())
                    .that(solution.plusOne(sample.input.digits))
                    .asList()
                    .containsExactlyElementsIn(
                        Arrays
                            .stream(sample.output)
                            .boxed()
                            .toList()
                    );
            }
        }
    }
}

File: kotlin/src/test/kotlin/problems1_100/PlusOneTest.kt
package problems1_100

import com.google.common.truth.Truth.assertWithMessage
import sample.SampledTest
import sample.component1
import sample.component2
import sample.component3
import sample.getSamples
import kotlin.test.Test

class PlusOneTest : SampledTest() {
    @Test
    fun test() =
        PlusOne.entries.forEach { solution ->
            getSamples<Array<PlusOneSample>>().forEach { (input, output, message) ->
                assertWithMessage(message)
                    .that(solution.plusOne(input.digits))
                    .asList()
                    .containsExactlyElementsIn(output.toList())
                    .inOrder()
            }
        }
}
